libs/ledger-live-common/src/wallet-api/react.ts

```diff
@@ -103,3 +147,521 @@ export function useGetAccountIds(
     }, new Map());
   }, [accounts, accounts$]);
 }
+
+export interface UiHook {
+  "account.request": (params: {
+    accounts$: Observable<WalletAPIAccount[]>;
+    currencies: CryptoCurrency[];
+    onSuccess: (
+      account: AccountLike,
+      parentAccount: Account | undefined
+    ) => void;
+    onError: () => void;
+  }) => void;
+  "account.receive": (params: {
+    account: AccountLike;
+    parentAccount: Account | undefined;
+    accountAddress: string;
+    onSuccess: (address: string) => void;
+    onCancel: () => void;
+    onError: (error: Error) => void;
+  }) => void;
+  "message.sign": (params: {
+    account: AccountLike;
+    message: MessageData | TypedMessageData;
+    onSuccess: (signature: string) => void;
+    onError: (error: Error) => void;
+    onCancel: () => void;
+  }) => void;
+  "transaction.sign": (params: {
+    account: AccountLike;
+    parentAccount: Account | undefined;
+    signFlowInfos: {
+      canEditFees: boolean;
+      hasFeesProvided: boolean;
+      liveTx: Partial<Transaction>;
+    };
+    options: Parameters<WalletHandlers["transaction.sign"]>[0]["options"];
+    onSuccess: (signedOperation: SignedOperation) => void;
+    onError: (error: Error) => void;
+  }) => void;
+  "transaction.broadcast": (
```

Comment:
Shouldn't it be `"transaction.signAndBroadcast"`?

`transaction.broadcast` does not exist anymore and has been replaced by `transaction.signAndBroadcast`
cf. [here](https://github.com/LedgerHQ/wallet-api/blob/main/packages/client/src/modules/Transaction.ts)

Reply:
In short, `keyof UIHook != keyof WalletHandlers`.

`transaction.signAndBroadcast` uses `transaction.sign` + `transaction.broadcast` UI hook underneath. The reason why it isn't called `signAndBroadcast` is that `sign` and `broadcast` require different UI to trigger.

Reply:
In short, keyof UIHook != keyof WalletHandlers.

transaction.signAndBroadcast uses transaction.sign + transaction.broadcast UI hook underneath. The reason why it isn't called signAndBroadcast is that sign and broadcast require different UI to trigger.